airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbDestinationStrictEncrypt.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.dynamodb;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DynamodbDestinationStrictEncrypt extends DynamodbDestination {
+
+  private static final String NON_SECURE_URL_ERR_MSG = "Server Endpoint requires HTTPS";
+  private static final Logger LOGGER = LoggerFactory.getLogger(DynamodbDestinationStrictEncrypt.class);
+
+  public DynamodbDestinationStrictEncrypt() {
+    super();
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) {
+    try {
+      final DynamodbDestinationConfig dynamodbDestinationConfig =
+          DynamodbDestinationConfig.getDynamodbDestinationConfig(config);
+
+      // enforce ssl connection
+      if (!DynamodbChecker.testCustomEndpointSecured(dynamodbDestinationConfig.getEndpoint())) {
+        return new AirbyteConnectionStatus()
+            .withStatus(AirbyteConnectionStatus.Status.FAILED)
+            .withMessage(NON_SECURE_URL_ERR_MSG);
+      }
+
+      DynamodbChecker.attemptDynamodbWriteAndDelete(dynamodbDestinationConfig);
```

Comment:
Nit: this seems redundant since you call `super.check(config)` outside the try/catch block since that already does this validation method

A suggestion would be to have the code like

```java
@Override
  public AirbyteConnectionStatus check(final JsonNode config) {
    final S3DestinationConfig destinationConfig = this.configFactory.getS3DestinationConfig(config, super.storageProvider());

    // enforce ssl connection
    if (!DynamodbChecker.testCustomEndpointSecured(destinationConfig.getEndpoint())) {
      return new AirbyteConnectionStatus()
                .withStatus(Status.FAILED)
          .withMessage("NON_SECURE_URL_ERR_MSG");
    }
    return super.check(config);
  }
```

The try/catch block isn't necessary since the method `testCustomEndpointSecured` is defined by you and it doesn't throw any `Exceptions` and `DynamodbDestination` already covers the try/catch within it's `check()` operation

Reply:
Good catch, updated, thanks!